common/types.go

```diff
@@ -50,3 +41,22 @@ func (h Hash) ToBytes() []byte {
 	b := [32]byte(h)
 	return b[:]
 }
+
+// BlockHeader is the header of a Polkadot block
+type BlockHeader struct {
+	ParentHash     Hash     // the block hash of the block's parent
+	Number         *big.Int // block number
+	StateRoot      Hash     // the root of the state trie
+	ExtrinsicsRoot Hash     // the root of the extrinsics trie
+	Digest         []byte   // any additional block info eg. logs, seal
+}
+
+type OptionalUint32 struct {
```

Comment:
These should probably be moved to a `common/` sub-package? The value/exists should be private and accessible with getters to avoid mutation. Probably also needs a `Set()` method

Reply:
added them to a subpackage, now we have `optional.Uint32` and `optional.NewUint32` to create a new one